be/src/Project/Web/Buyer/Buyer.Web/Shared/Repositories/MainNavigationLinks/MainNavigationLinkRepository.cs

```diff
@@ -0,0 +1,85 @@
+﻿using Buyer.Web.Shared.DomainModels.MainNavigationLinks;
+using Buyer.Web.Shared.GraphQlResponseModels;
+using Foundation.Extensions.ExtensionMethods;
+using GraphQL;
+using GraphQL.Client.Abstractions;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace Buyer.Web.Shared.Repositories.MainNavigationLinks
+{
+    public class MainNavigationLinkRepository : IMainNavigationLinkRepository
+    {
+        private readonly IGraphQLClient _graphQlClient;
+        private readonly ILogger<MainNavigationLinkRepository> _logger;
+
+        public MainNavigationLinkRepository(
+            IGraphQLClient graphQlClient,
+            ILogger<MainNavigationLinkRepository> logger)
+        {
+            _graphQlClient = graphQlClient;
+            _logger = logger;
+        }
+
+        public async Task<IEnumerable<MainNavigationLink>> GetMainNavigationLinksAsync(string contentPageKey, string language)
+        {
+            try
+            { 
+                var response = await _graphQlClient.SendQueryAsync<JObject>(GetMainNavigationLinksQuery(contentPageKey, language));
+
+                if (response.Errors.OrEmptyIfNull().Any() is false && response?.Data != null)
```

Comment:
**suggestion:** Missing language fallback. If you don't know how, take a look at HeroSliderRepository